### PR TITLE
BUG: support empty rows in `Metadata.load`

### LINKED ISSUE #336: Metadata with blank rows difficult to diagnose
Metadata with empty rows:
screen shot 2017-09-21 at 6 53 18 pm

Validation exception, while technically true, doesn't really help solve the problem.
```bash
Traceback (most recent call last):
  File "/Users/matthew/miniconda3/envs/q2dev-2017.9/bin/qiime", line 11, in <module>
    load_entry_point('q2cli', 'console_scripts', 'qiime')()
  File "/Users/matthew/miniconda3/envs/q2dev-2017.9/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/matthew/miniconda3/envs/q2dev-2017.9/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/matthew/miniconda3/envs/q2dev-2017.9/lib/python3.5/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/matthew/miniconda3/envs/q2dev-2017.9/lib/python3.5/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/matthew/miniconda3/envs/q2dev-2017.9/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/matthew/miniconda3/envs/q2dev-2017.9/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/matthew/caporaso/qiime2/q2cli/q2cli/commands.py", line 189, in __call__
    arguments, missing_in, verbose, quiet = self.handle_in_params(kwargs)
  File "/Users/matthew/caporaso/qiime2/q2cli/q2cli/commands.py", line 281, in handle_in_params
    kwargs, fallback=cmd_fallback
  File "/Users/matthew/caporaso/qiime2/q2cli/q2cli/handlers.py", line 378, in get_value
    metadata.append(qiime2.Metadata.load(path))
  File "/Users/matthew/miniconda3/envs/q2dev-2017.9/lib/python3.5/site-packages/qiime2/metadata.py", line 120, in load
    return cls(df)
  File "/Users/matthew/miniconda3/envs/q2dev-2017.9/lib/python3.5/site-packages/qiime2/metadata.py", line 55, in __init__
    _validate_pandas_index(dataframe.index, 'index')
  File "/Users/matthew/miniconda3/envs/q2dev-2017.9/lib/python3.5/site-packages/qiime2/metadata.py", line 29, in _validate_pandas_index
    raise ValueError(invalid_metadata_template % msg)
ValueError: Non-string Metadata index values detected. There may be more errors present in this metadata. Currently only QIIME 1 sample/feature metadata mapping files are officially supported. Sample metadata files can be validated using Keemei: http://keemei.qiime.org.
```

It would be nice to check specifically for blank rows and report back on that to the user.

This recently came up on [the forum](https://forum.qiime2.org/t/error-in-command-for-generating-pcoa-plot/1251?u=thermokarst).

### PR DESCRIPTION
**Pull request by jairideout**
Empty rows are now ignored, similar to how blank lines and comments are ignored. In this case, an "empty row" is a row consisting only of empty cells (i.e. only tab characters).

Fixes #336.

### CODE REVIEW

File: qiime2/tests/data/metadata/empty-rows.tsv [hunk 1]
```diff
@@ -0,0 +1,8 @@
+my-index	col1	col2	col3
+id1	1	a	foo
```
Comment (antgonza) [inline qiime2/tests/data/metadata/empty-rows.tsv]: Excel tends to add also spaces (not sure why or how) to the empty rows/cols, could you add these 2 to your test? Basically, add another "blank" column/row with empty spaces.

**Comment by jairideout** [comment]
Thanks for reviewing @antgonza! I addressed your comment in a squash commit.
